src/types/execution_result.rs

```diff
@@ -0,0 +1,72 @@
+use pyo3::{pyclass, pymethods};
+use revm::primitives::{ExecutionResult as RevmExecutionResult, HaltReason, OutOfGasError, SuccessReason};
+
+// pub struct Log(RevmLog);
+
+
+#[derive(Debug)]
+#[pyclass(get_all)]
+pub struct ExecutionResult {
+    is_success: bool,
+    is_halt: bool,
+    reason: String,
+    gas_used: u64,
+    gas_refunded: u64,
+    // TODO: logs: Vec<Log>,
+}
+
+#[pymethods]
+impl ExecutionResult {}
+
+impl From<RevmExecutionResult> for ExecutionResult {
+    fn from(result: RevmExecutionResult) -> Self {
+        ExecutionResult {
+            is_success: result.is_success(),
+            is_halt: result.is_halt(),
+            reason: match result {
+                RevmExecutionResult::Success { reason, .. } => match reason {
+                    SuccessReason::Stop => String::from("Stop"),
+                    SuccessReason::Return => String::from("Return"),
+                    SuccessReason::SelfDestruct => String::from("SelfDestruct"),
+                }
+                RevmExecutionResult::Revert { .. } => String::from("Revert"),
+                RevmExecutionResult::Halt { reason, .. } => match reason {
+                    HaltReason::OutOfGas(out_of_gas) => match out_of_gas {
+                        OutOfGasError::Basic => String::from("OutOfGas:Basic"),
```

Comment:
is there any kind of macro to do this?

Reply:
I guess this can mostly be `format!("{result:?}")` (using debug format `:?` prints just the name of the variant)